repose-aggregator/services/atom-feed/impl/src/main/resources/META-INF/schema/config/atom-feed-service.xsd

```diff
@@ -0,0 +1,183 @@
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+
+        <xs:attribute name="entry-order" type="EntryOrderType" use="optional" default="update">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        Defines the order in which new Atom events will be delivered to any listeners registered
+                        on this feed.
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:simpleType name="EntryOrderType">
+        <xs:restriction base="xs:string">
+            <xs:enumeration value="random">
+                <xs:annotation>
+                    <xs:documentation>
+                        <html:p>
+                            With this ordering, a feed listener will receive events with no ordering guarantees.
+                        </html:p>
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:enumeration>
+
+            <xs:enumeration value="update">
+                <xs:annotation>
+                    <xs:documentation>
+                        <html:p>
+                            With this ordering, a feed listener will receive events in the order in which they were
+                            updated. That is, entries will be ordered chronologically based on their updated field
+                            from oldest to newest.
+                        </html:p>
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:enumeration>
+
+            <!--xs:enumeration value="publish">
+                <xs:annotation>
+                    <xs:documentation>
+                        <html:p>
+                            With this ordering, a feed listener will receive events in the order in which they were
+                            published.
+                        </html:p>
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:enumeration-->
+        </xs:restriction>
+    </xs:simpleType>
+
+    <xs:simpleType name="PositiveIntType">
+        <xs:restriction base="xs:int">
+            <xs:minInclusive value="1"/>
+        </xs:restriction>
+    </xs:simpleType>
+
+    <xs:complexType name="AuthenticationType">
+        <xs:attribute name="fqcn" type="xs:string"
+                      default="org.openrepose.nodeservice.atomfeed.impl.auth.NoopAuthenticatedRequestFactory">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>Defines the fully qualified class name of a class which can authenticate requests.</html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="OpenStackIdentityFQCNType">
```

Comment:
Thinking about this now, it seems this feat is actually quite a bit more challenging for the user than expected. Let us posit that the user does not have access to this schema (since it is hidden away in an artifact somewhere). Then that user cannot extend a type from this schema. The parser inside of Repose will always attempt to use this schema to validate a configuration file, and the user is unable to change this schema.

So I've come up with the following for now: allow any element or attribute to be attached to the AuthenticationType. The result is that the AuthenticationType can pass any data in any XML-supported format to the code. Unfortunately, the Repose parser will not be able to validate any of the elements or attributes which are not explicitly defined in this schema. This means that data can be passed around, but not validated like the rest of the configuration.

I believe there may be a way to have the XML configuration file tell the parser what to validate and where the validation rules live, but I haven't been able to figure that part out yet. To be more specific, the user would probably define a type which extends the AuthenticationType in their own XSD which sits alongside their configuration XML. Of course, doing so would require that their XSD be aware of the AuthenticationType, which probably relies on the parser to bridge that gap. Once the XSD is written, the user's configuration file would have to declare the type on their AuthenticationType element as I've done with the OpenStack Identity type. Doing so requires that the parser know where that type lives, which is in the user-defined XSD. If the parser is smart enough to do all of that, type-safe extensibility from the user perspective may be possible.